apis/v1alpha1/tcproute_types.go

```diff
@@ -51,13 +51,15 @@ type TCPRoute struct {
 
 // TCPRouteRule is the configuration for a given rule.
 type TCPRouteRule struct {
-	// Match defines which connections match this rule.
+	// Matches define conditions used for matching the rule against
+	// incoming TCP connections.
+	// Each match is independent, i.e. this rule will be matched
+	// if **any** one of the matches is satisfied.
 	//
 	// +optional
-	Match *TCPRouteMatch `json:"match" protobuf:"bytes,1,opt,name=match"`
+	Matches []TCPRouteMatch `json:"matches" protobuf:"bytes,1,rep,name=matches"`
+
 	// Action defines what happens to the connection.
-	//
-	// +optional
```

Comment:
Instead of just removing `+optional` we probably want to make this non-nil and required.